include/exec/task.hpp

```diff
@@ -141,22 +231,37 @@ namespace exec {
       requires stdexec::unstoppable_token<
         stdexec::stop_token_of_t<stdexec::env_of_t<_ParentPromise>>>
     struct __default_awaiter_context<_ParentPromise> {
-      explicit __default_awaiter_context(__default_task_context_impl&, _ParentPromise&) noexcept {
+      template <__scheduler_affiinty _A>
+      explicit __default_awaiter_context(
+        __default_task_context_impl<_A>& __self,
+        _ParentPromise& __parent) noexcept {
+        if constexpr (
+          __indirect_completion_scheduler_provider<_ParentPromise>
+          && _A == __scheduler_affiinty::__sticky) {
+          __self.__scheduler_ = stdexec::get_completion_scheduler<stdexec::set_value_t>(
+            stdexec::get_env(__parent));
+        }
       }
     };
 
     // Finally, if we don't know the parent coroutine's promise type, assume the
     // worst and save a type-erased stop callback.
     template <>
     struct __default_awaiter_context<void> {
-      template <class _Ty>
-      explicit __default_awaiter_context(__default_task_context_impl&, _Ty&) noexcept {
+      template <__scheduler_affiinty _A, class _Ty>
+      explicit __default_awaiter_context(__default_task_context_impl<_A>&, _Ty&) noexcept {
       }
 
-      template <__indirect_stop_token_provider _ParentPromise>
+      template <__scheduler_affiinty _A, __indirect_stop_token_provider _ParentPromise>
```

Comment:
typo here, affinity instead of affiinty.

Reply:
thank you <3

Reply:
Fixed